modules/Core/pages/validate.php

```diff
@@ -29,9 +29,124 @@
             $user,
         ));
 
-        Session::flash('home', $language->get('user', 'validation_complete'));
+        if (Session::exists('validate_email')) {
+            Session::delete('validate_email');
+        }
+
+        Session::flash('login_success', $language->get('user', 'validation_complete'));
+        Redirect::to(URL::build('/login'));
     } else {
         Session::flash('home_error', $language->get('user', 'validation_error'));
     }
+
+    Redirect::to(URL::build('/'));
+} else if (Session::exists('validate_email')) {
+    $email_address = Session::get('validate_email');
+
+    $target_user = new User($email_address, 'email');
+    if (!$target_user->exists() || $target_user->data()->active) {
+        Redirect::to(URL::build('/'));
+    }
+
+    // Handle input
+    if (Input::exists()) {
+        if (Token::check()) {
+            $validation = Validate::check($_POST, [
+                'email' => [
+                    Validate::REQUIRED => true,
+                    Validate::EMAIL => true,
+                    Validate::UNIQUE => ['users', 'id:' . $target_user->data()->id],
+                    Validate::RATE_LIMIT => [1, 3600]
+                ]
+            ])->messages([
+                'email' => [
+                    Validate::REQUIRED => $language->get('user', 'email_required'),
+                    Validate::EMAIL => $language->get('general', 'contact_message_email'),
+                    Validate::UNIQUE => $language->get('user', 'email_already_exists'),
+                    Validate::RATE_LIMIT => static fn($meta) => $language->get('general', 'rate_limit', $meta)
```

Comment:
does this term exist?

Reply:
Fixed, I just copyed the validation from settings page so was same issue there